### DISCUSSION TITLE
StageOneReportTests and question on assertFileHasContent

### DISCUSSION THREAD

**Discussion by arif-ali** [sosreport/sos#3836] [Q&A] [answered]
I may have been silent on the repo here, but have been working in the background to do stageone product tests for both sunbeam (whch is next generation Canonical Openstack), and then devstack, especially after the long discussions in #3622

The initial phase of sunbeam is ready, and now got a point where I have 5 failures, and all ware related to self.assertFileHasContent()

Now, it seems, if it finds the pattern more then once, it will cause a failure, is this intended, as I am looking for `password = *********`, and that would appear twice in the file we are obfuscated.

If this is the intention, then I have to re-think in how we do these tests.

Any clarification on this would be awesome

**Comment by TurboTurtle** [comment, marked as answer]
Not the intent, no. We do currently issue a `break` once we find the content however which may be inadvertently causing your failures.

We could perhaps have it re-written to parse the entire file as a whole instead of a line-by-line check.

**Reply by arif-ali** [reply]
cool, I'll see if I can evaluate an update to the function, so that we don't error out, and maybe it can output how many times it saw that content
